wagtail/contrib/simple_translation/views.py

```diff
@@ -137,7 +137,11 @@ def get_object(self):
         if not issubclass(model, TranslatableMixin):
             raise Http404
 
-        return get_object_or_404(model, pk=unquote(self.kwargs["pk"]))
+        object = get_object_or_404(model, pk=unquote(self.kwargs["pk"]))
+        if isinstance(object, DraftStateMixin):
+            object = object.get_latest_revision_as_object()
```

Comment:
This doesn't really affect how the object is translated (that is done in `CopyForTranslationAction`), but this makes sure anything shown from the instance (e.g. `str(object)`) uses the latest revision